packages/@glimmer/compiler/lib/compiler.ts

```diff
@@ -116,10 +112,20 @@ export function precompile(
     id: idFn(JSON.stringify(options.meta) + blockJSON),
     block: blockJSON,
     moduleName: moduleName ?? '(unknown template module)',
+    scope: null,
+    isStrictMode: options.strictMode ?? false,
   };
 
   // JSON is javascript
-  return JSON.stringify(templateJSONObject);
+  let stringified = JSON.stringify(templateJSONObject);
+
+  if (usedLocals.length > 0) {
```

Comment:
Should this be something like:

```suggestion
  if (options.strictMode && usedLocals.length > 0) {
```

?

Reply:
yeah, that makes sense. In theory `usedLocals` should only exist if we are in strict mode, but it's better to be explicit I think.